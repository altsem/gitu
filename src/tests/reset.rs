use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    commit(ctx.dir.path(), "unwanted-file", "");
    ctx
}

#[test]
pub(crate) fn reset_menu() {
    snapshot!(setup(), "lljX");
}

#[test]
pub(crate) fn reset_soft_prompt() {
    snapshot!(setup(), "lljXsq");
}

#[test]
pub(crate) fn reset_soft() {
    snapshot!(setup(), "lljXs<enter>q");
}

#[test]
pub(crate) fn reset_mixed() {
    snapshot!(setup(), "lljXm<enter>q");
}

#[test]
fn reset_hard() {
    snapshot!(setup(), "lljXh<enter>q");
}
