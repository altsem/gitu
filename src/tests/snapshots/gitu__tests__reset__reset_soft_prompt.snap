---
source: src/tests/reset.rs
expression: ctx.redact_buffer()
---
 ba1a85d main add unwanted-file                                                 |
▌b66a0bf origin/main add initial-file                                           |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
? Soft reset to (default b66a0bf82020d6a386e94d0fceedec1f817d20c7): › q         |
styles_hash: 174798f428e40bd6
